File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.java
class Solution {
    public int[][] largestLocal(int[][] grid) {

        int n = grid.length;
        int m = n - 2;
        
        int ans [][] = new int [m][m];

        for(int i = 0; i < m; i++){
            
            for(int j = 0; j < m; j++){
                ans[i][j] = getMax(grid, i, j);
            }
        }
        return ans;
    }

 int getMax(int [][] grid, int i, int j){
        int max = 0;
        for(int row = i; row < i+3; row++){
            for(int col = j; col < j+3; col++){
                max = Math.max(max, grid[row][col]);
            }
        }
        return max;
    }
}

File: 2454-largest-local-values-in-a-matrix/largest-local-values-in-a-matrix.js
/**
 * @param {number[][]} grid
 * @return {number[][]}
 */
var largestLocal = function (grid) {
    let n = grid.length;
    let m = n - 2;

    let ans = [];

    for (let i = 0; i < m; i++) {
        let tm = [];
        for (let j = 0; j < m; j++) {
            tm.push(getMax(grid, i, j));
        }
        ans.push(tm);
    }
    return ans;

};
function getMax(grid, i, j) {
    let max = 0;
    for (let row = i; row < i + 3; row++) {
        for (let col = j; col < j + 3; col++) {
            max = Math.max(max, grid[row][col]);
        }
    }
    return max;
}